scripts/photon-containers.sh

```diff
@@ -4,52 +4,42 @@
 
 echo '> Pre-Downloading Kubeadm Docker Containers'
 
-CONTAINERS=(
-k8s.gcr.io/kube-apiserver:v1.14.9
-k8s.gcr.io/kube-controller-manager:v1.14.9
-k8s.gcr.io/kube-scheduler:v1.14.9
-k8s.gcr.io/kube-proxy:v1.14.9
-k8s.gcr.io/pause:3.1
-k8s.gcr.io/etcd:3.3.10
-k8s.gcr.io/coredns:1.3.1
-antrea/antrea-ubuntu:v0.6.0
-embano1/tinywww:latest
-projectcontour/contour:v1.0.0-beta.1
-openfaas/faas-netes:0.9.0
-openfaas/gateway:0.17.4
-openfaas/basic-auth-plugin:0.17.0
-openfaas/queue-worker:0.8.0
-openfaas/faas-idler:0.2.1
-envoyproxy/envoy:v1.11.1
-prom/prometheus:v2.11.0
-prom/alertmanager:v0.18.0
-nats-streaming:0.11.2
-vmware/veba-event-router:${VEBA_VERSION}
-)
+VEBA_BOM_FILE=/root/config/veba-bom.json
 
-for i in ${CONTAINERS[@]};
+for component_name in $(jq '. | keys | .[]' ${VEBA_BOM_FILE});
 do
-	docker pull $i
+    if [ $component_name != "\"veba\"" ]; then
+        for i in $(jq ".$component_name.containers | keys | .[]" ${VEBA_BOM_FILE}); do
```

Comment:
This assumes that, besides `"veba"` all other components will have a `containers` array. Is this a reasonable assumption, especially if we later want to add more keys potentially without `containers`?  Is there a way to tell jq to not error if a key like `.containers` does not exist and just skip?

Reply:
Addressed by checking whether .containers property exists